### PR TITLE
Specializing DSS versions

### PR DESCRIPTION
**Pull request by sriharshakandala**
Specialize DSS for various Spaces.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Spaces/dss.jl [hunk 1]
```diff
@@ -612,29 +612,60 @@ function dss_2d!(
     return data
 end
 
+function weighted_dss!(
+    data,
+    hspace::SpectralElementSpace2D,
+    ghost_buffer = nothing,
+)
+    @show "calling SpectralElementSpace2D DSS"
```
Comment (simonbyrne) [inline src/Spaces/dss.jl:620]: get rid of the `@show`s

**Comment by sriharshakandala** [comment]
bors try

File: src/Spaces/dss.jl [hunk 2]
```diff
@@ -612,29 +612,57 @@ function dss_2d!(
...
 
-function weighted_dss!(data, space::AbstractSpace, ghost_buffer = nothing)
-    if space isa ExtrudedFiniteDifferenceSpace
-        hspace = space.horizontal_space
-    else
-        hspace = space
-    end
+function weighted_dss!(
+    data,
+    space::ExtrudedFiniteDifferenceSpace{S, H},
```
Comment (simonbyrne) [inline src/Spaces/dss.jl:642]: Although it's not used, it still needs to be there for compatibility:
```suggestion
    space::ExtrudedFiniteDifferenceSpace{S, H},
    ghost_buffer = nothing,
```

**Comment by sriharshakandala** [comment]
bors try

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**